ql/src/java/org/apache/hadoop/hive/ql/exec/repl/ReplDumpTask.java

```diff
@@ -198,7 +223,12 @@ private void initiateDataCopyTasks() throws SemanticException {
       finishRemainingTasks();
     } else {
       DAGTraversal.traverse(childTasks, new AddDependencyToLeaves(TaskFactory.get(work, conf)));
-      this.childTasks = childTasks;
```

Comment:
If Ranger dump fails, the task execution will be stopped there itself with a failure and we will get an error. It won't even come to the second ReplDump task execution to execute the finishRemainingTasks.

Reply:
Let's say, during an incremental run, you don't have any new table to bootstrap and you don't have even any external table in DB.
List<Task<?>> childTasks = new ArrayList<>();
    childTasks.addAll(work.externalTableCopyTasks(taskTracker, conf));
    childTasks.addAll(work.managedTableCopyTasks(taskTracker, conf));
    if (childTasks.isEmpty()) {
      //All table data copy work finished.
      finishRemainingTasks();
    }
This will go to finishRemainingTasks() immediately even though you have rangerDumpTask as childTask of ReplDumpTask